src/data/adapter.h

```diff
@@ -234,22 +234,22 @@ class ArrayAdapterBatch : public detail::NoMetaInfo {
   class Line {
     ArrayInterface array_interface_;
     size_t ridx_;
+
    public:
     Line(ArrayInterface array_interface, size_t ridx)
         : array_interface_{std::move(array_interface)}, ridx_{ridx} {}
 
     size_t Size() const { return array_interface_.num_cols; }
 
     COOTuple GetElement(size_t idx) const {
-      return {ridx_, idx, array_interface_.GetElement(idx)};
+      return {ridx_, idx, array_interface_.GetElement(ridx_, idx)};
     }
   };
 
  public:
   ArrayAdapterBatch() = default;
   Line const GetLine(size_t idx) const {
-    auto line = array_interface_.SliceRow(idx);
```

Comment:
@hcho3 Previously we extract a row for the `Line`, now we just pass the whole array interface into each line.

Reply:
@trivialfis Ah everything makes sense now. Thanks for taking time to explain.